tests/boards/intel_adsp/smoke/src/main.c

```diff
@@ -42,6 +42,30 @@ ZTEST(intel_adsp, test_clock_calibrate)
 	zassert_true((hz / MIN(1, diff)) > 100, "clock rate wrong");
 }
 
+#if XCHAL_HAVE_VECBASE
+ZTEST(intel_adsp, test_vecbase_lock)
+{
+	uintptr_t vecbase;
+
+	/* Unfortunately there is not symbol to check if the target
+	 * supports locking VECBASE. The best we can do is checking if
+	 * the first is not set and skip the test.
+	 */
+	__asm__ volatile("rsr.vecbase %0" : "=r"(vecbase));
+	if ((vecbase & 0x1) == 0) {
+		ztest_test_skip();
+	}
+
+	/* VECBASE register should have been locked during the cpu
+	 * start up. Trying to change its location should fail.
+	 */
+	__asm__ volatile("wsr.vecbase %0; rsync" : : "r"(0x0));
```

Comment:
FWIW, if the target does support writes to the vector table, and you happen to take an interrupt while this NULL is sitting in the register, the world will blow up.   I get that this represents a novel way to detect what is otherwise a failure condition anyway, but general hygiene would argue that this should be wrapped in an arch_irq_lock/unlock() (or just a spinlock, etc...)

Reply:
That is about locking VECBASE, which is done during the initialization, this means that attempting to change VECBASE will fail. If this is not locked yes ... it will be problematic, the test can potentially crash. Test will fail anyways.

Reply:
No no, I get what it's doing.  I'm just saying that getting a clean "FAIL" on output is a cleaner failure than a spurious device lockup, which can look like a different kind of failure entirely.  Just mask interrupts, it's the right thing to do.